### DISCUSSION TITLE
NextJs image optimization not working in Vercel production (400: BAD_REQUEST) with @payloadcms/storage-uploadthing

### DISCUSSION THREAD

**Discussion by poofy25** [payloadcms/payload#6704] [Q&A] [answered]
### Payload Version

3.0.0-beta.36

### Node Version

v21.4.0

### Next.js Version

15.0.0-rc.0

### Describe the Bug

I can't fetch images from the local api and display them in an <Image/> component

In dev mode everything works fine
`
{
...
"url": "/api/media/file/137-92091.jpeg",
...
}
`
I get the url from the media collection files and then display it in the `<Image/>` component as follows 

`
<Image
        className="h-full w-[50%] object-cover"
        src={blockData.image.url}
        alt="img"
        width="400"
        height="400"
      />
`

And it works just fine. The `<img/>` element then optimizes the image and returns this url
`/_next/image?url=%2Fapi%2Fmedia%2Ffile%2F137-92091.jpeg&w=640&q=75` 
and it works just fine in development mode.

But when i deployed it to Vercel when going to [https://starbucks-clone-orpin.vercel.app/api/media/file/137-92091.jpeg](url) the image loads but when the `<Image/>` component optimizes the image with the following url : 
`/_next/image?url=%2Fapi%2Fmedia%2Ffile%2F137-92091.jpeg`
it returns the following:
`
400: BAD_REQUEST
Code: INVALID_IMAGE_OPTIMIZE_REQUEST
ID: fra1::n89vt-1717417920074-c8806297edac
`

![image](https://github.com/payloadcms/payload/assets/117314241/81a68c1c-f677-4070-a8a6-4cd5f75b55fe)
![image](https://github.com/payloadcms/payload/assets/117314241/4fff33d7-6317-453e-9d38-d4528bc604e2)

### Reproduction Steps

Repository : [https://github.com/poofy25/starbucks-clone](url)

1. Clone the repository

2. Create .env file using the .env.example file 

3. Deploy to Vercel

### Adapters and Plugins

@payloadcms/storage-uploadthing

**Comment by laamdev** [comment]
I'm getting the same error when trying to display images in a Vercel deployment using a Supabase S3 bucket. Any solutions?

**Comment by poofy25** [comment, marked as answer]
> I'm getting the same error when trying to display images in a Vercel deployment using a Supabase S3 bucket. Any solutions?

So i found a workaround for this issue all you gotta do is add the website url in the image src path
For example 

Before:
<Image src={image.url} ...>

After 
<Image src={`${process.env.WEBSITE_URL}${image.url}`} />

**Reply by Dovchik** [reply]
giving absolute url doesn't works for me with Vercel blob: update - it works

**Comment by jmikrut** [comment]
Hey all, yes, as @poofy25 said, on your frontend, I believe you need to pass full URLs to the `next/image` component. I will convert this to a Discussion so that we can continue to help there!

**Reply by denisprado** [reply]
I am getting the same error.. you have any solution?

**Reply by JosephClay** [reply]
haven't had success using the full url recommendation. also, converting to a full url requires defining a remote pattern, which we shouldn't have to do on the same host (https://nextjs.org/docs/messages/next-image-unconfigured-host)

**Comment by abid-salahi** [comment]
This isn't working when using S3 bucket storage and Vercel deployment. Even using absolute urls.

**Comment by kotyperry** [comment]
Also getting this error still. @jmikrut Can we have someone relook into this?

**Comment by rslmrn** [comment]
Im also running with this issue, using absolute url does not fix the issue. Clicking and redirecting to the path it actually returns the image https://your-site/api/media/file/photo.png but when rendering in the next image getting the same error 400: BAD_REQUEST
Code: INVALID_IMAGE_OPTIMIZE_REQUEST
ID: sfo1::zr796-1724222305752-c64a4573adf7

**Comment by pnwatin** [comment]
Same problem here, having absolute url doesnt fix it. 
One temporary fix is to disable image optimization in nextjs image component with `unoptimized`

**Comment by denisprado** [comment]
I understand the argument to close the issue that the problem is with Next's image optimizer, but at the same time if Payload intends to have a default configuration that can be hosted on Vercel, the team needs to worry about this issue. Once there is a stable version available to a wider audience, this problem will make it impossible to use without headaches.

**Reply by KimPaow** [reply]
Agree with this, not really their issue, but this is such a huge dealbreaker that if I weren't already too invested and on a deadline I would opt out of using Payload because of this experience. If they want to give Next first-class support it's a requirement to find an easy to implement workaround.

**Comment by robertmalicke** [comment]
I agree. Also, for this to work, we have to add localhost to the next config allowed remote paths. This seems odd -- if they are running in the same sever, a FQDN should not be required. I believe this logic could be pushed down into Payload. I also agree this is a pretty significant thing for the masses who are about to roll into Payload from nextjs. They just need the image component to simply work. Absolute URL is also not working for me with next/image and vercel blob -- now I'm hard blocked on this!!

**Comment by cgaravitoq** [comment]
I was working with S3 bucket and Next.js 15 I just added "unoptimized" in the <Image/> component and it solved the issue 😶‍🌫️

**Reply by joaoantoniorabelo** [reply]
I have update my nextConfig where I set a configuration for images with a front-end loader.

Images were upload when deployed.

File: next.config.js

```
const nextConfig = {
  images: {
     loader: 'custom',
     loaderFile: 'path/loader.js',
     …originalObject
  }
}
```

File: loader.js
```
'use client'

export default function imageLoader({ src, width, quality }) {
 const url = new URL(src);

 let directPath = url.pathname;
   
    if (url.pathname.startsWith('/api/media/file')) {
        let path = url.pathname.split('file');
        directPath = `media${path[1]}`;
    }
    

    return `${url.protocol}//${url.host}/${directPath}?w=${width}&q=${quality || 75}`;
}
```

https://nextjs.org/docs/pages/api-reference/components/image#loader

**Comment by danioo** [comment]
Had the same issue, to fix it I had to define env variable (VERCEL_PROJECT_PRODUCTION_URL) for project in Vercel.

Payload website example uses NEXT_PUBLIC_SERVER_URL which depends on VERCEL_PROJECT_PRODUCTION_URL variable.

`const NEXT_PUBLIC_SERVER_URL = process.env.VERCEL_PROJECT_PRODUCTION_URL
  ? `https://${process.env.VERCEL_PROJECT_PRODUCTION_URL}`
  : undefined || process.env.NEXT_PUBLIC_SERVER_URL || 'http://localhost:3000'`

**Reply by joaoantoniorabelo** [reply]
I have applied the value and prioritized the VERCEL_PROJECT_PRODUCTION_URL, and still doesn't work for me. I had to use the imageloader workaround for now

**Comment by MacPiston** [comment]
This is still an issue - using absolute URL doesn't change a thing. Decoded image URL taken from rendered Next's Image `img` tag works just fine, and correct image is returned.
